internal/loader/loader.go

```diff
@@ -79,13 +87,48 @@ func (l *Loader) Load(ctx context.Context, warcId string) (gowarc.WarcRecord, er
 				Date:      record.WarcHeader().Get(gowarc.WarcRefersToDate),
 			}
 		}
+
+		var revisitOf gowarc.WarcRecord
 		storageRef, err = l.Resolver.Resolve(warcRefersTo)
-		if err != nil {
+		// if the record is missing from out DB and a proxy is configured, then we should
+		// ask the proxy to get the revisitOf record for us
+		if errors.Is(err, badger.ErrKeyNotFound) && l.ProxyUrl != nil {
+			reqUrl := *l.ProxyUrl
+			reqUrl.Path = path.Join(reqUrl.Path, "id", warcRefersTo)
+
+			log.Debug().Msgf("attempt to get record from proxy url %s", reqUrl.String())
+			req, err := http.NewRequestWithContext(ctx, "GET", reqUrl.String(), nil)
+			if err != nil {
+				return nil, err
+			}
+
+			resp, err := http.DefaultClient.Do(req)
```

Comment:
Unecessary to create http.Client every time. Should be package global variable (see client internal/server/handlers/handlers.go for example).

Reply:
Using default client does not create a new client as it is a http package wide client pointer:

client.go:
```go
// DefaultClient is the default Client and is used by Get, Head, and Post.
var DefaultClient = &Client{}
```